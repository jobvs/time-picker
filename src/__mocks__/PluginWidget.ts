export function createEditableValue<T>(
    value: T,
    readOnly: boolean,
    validation: string[]
): PluginWidget.EditableValue<T> {
    return {
        value,
        displayValue: "display",
        status: PluginWidget.ValueStatus.Available,
        validation,
        readOnly,
        formatter: null,
        universe: [],
        setValue: () => null,
        setTextValue: () => null,
        setFormatting: () => null,
        setValidation: () => null
    };
}

export function createDynamicValue<T>(value: T): PluginWidget.DynamicValue<T> {
    return {
        value,
        status: PluginWidget.ValueStatus.Available
    };
}

export function createActionValue(execute: () => void): PluginWidget.ActionValue {
    return {
        canExecute: true,
        isExecuting: false,
        execute
    };
}
