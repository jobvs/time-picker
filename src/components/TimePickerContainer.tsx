import * as React from "react";
import { hot } from "react-hot-loader";

import Label from "./Label";
import TimePicker from "./TimePicker";

export interface TimePickerContainerProps {
  class: string;
  style: React.CSSProperties;

  showLabel: boolean;
  labelCaption: string;
  labelOrientation: "horizontal" | "vertical";
  labelWidth: number;

  tabIndex: number;
  name: string;
  format: string;
  inputValue: PluginWidget.EditableValue<Date>;
  placeholder: PluginWidget.DynamicValue<string>;
  onChange?: PluginWidget.ActionValue;
}

export class TimePickerContainer extends React.Component<TimePickerContainerProps> {
  render() {
    return (
      <div className={this.props.class} style={this.props.style}>
        {this.props.showLabel
          ? this.renderTimePickerWithLabel()
          : this.renderTimePicker()}
      </div>
    );
  }

  renderTimePickerWithLabel() {
    return (
      <Label
        label={this.props.labelCaption}
        orientation={this.props.labelOrientation}
        width={this.props.labelWidth}
      >
        {this.renderTimePicker()}
      </Label>
    );
  }

  renderTimePicker() {
    return (
      <TimePicker
        tabIndex={this.props.tabIndex}
        name={this.props.name}
        format={this.props.format}
        inputValue={this.props.inputValue}
        placeholder={this.props.placeholder}
        onChange={this.props.onChange}
      />
    );
  }
}

export default hot(module)(TimePickerContainer);
