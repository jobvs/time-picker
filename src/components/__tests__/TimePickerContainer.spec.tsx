import React from "react";
import renderer from "react-test-renderer";

import TimePickerContainer, { TimePickerContainerProps } from "../TimePickerContainer";
import { createActionValue, createDynamicValue, createEditableValue } from "../../__mocks__/PluginWidget";

const setup = (propOverrides: Partial<TimePickerContainerProps>) => {
  const props: TimePickerContainerProps = {
    inputValue: createEditableValue(new Date(1544793796784), false, []),
    editable: "default",
    timeFormat: "milliseconds",
    timeNotation: "h12",
    placeholder: createDynamicValue("Placeholder"),
    showLabel: false,
    labelCaption: "Default caption",
    labelOrientation: "vertical",
    labelWidth: 3,
    name: "custom-name",
    class: "custom-class",
    style: { backgroundColor: "transparent" },
    tabIndex: 0,
    hoursStep: 0,
    minutesStep: 0,
    secondsStep: 0,
    onChange: createActionValue(jest.fn),
    onEnter: createActionValue(jest.fn),
    onLeave: createActionValue(jest.fn),
    ...propOverrides
  };

  return renderer.create(<TimePickerContainer {...props} />);
};

describe("TimePicker", () => {
  it("should render the structure", () => {
    const tree = setup({}).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render without a data attribute", () => {
    const tree = setup({
      inputValue: undefined
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render 24-hour notation", () => {
    const tree = setup({
      timeNotation: "h24"
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe("label", () => {
    it("should render a vertical label", () => {
      const tree = setup({
        showLabel: true
      }).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should render a horizontal label", () => {
      const tree = setup({
        showLabel: true,
        labelOrientation: "horizontal"
      }).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should render a horizontal label with a specified width", () => {
      const tree = setup({
        showLabel: true,
        labelOrientation: "horizontal",
        labelWidth: 6
      }).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it("should render with a different caption", () => {
      const tree = setup({
        showLabel: true,
        labelCaption: "Different"
      }).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
