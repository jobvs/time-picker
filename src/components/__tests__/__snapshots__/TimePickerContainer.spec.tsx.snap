// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker label should render a horizontal label 1`] = `
<div
  className="custom-class"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="form-group clearfix "
  >
    <label
      className="control-label col-sm-3"
    >
      Default caption
    </label>
    <div
      className="col-sm-9"
    >
      <div
        className="widget-timepicker"
      >
        <div
          className="rdt"
        >
          <input
            className="form-control"
            disabled={false}
            name="custom-name"
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Placeholder"
            tabIndex={0}
            type="text"
            value="02:23:16.784 PM"
          />
          <div
            className="rdtPicker"
          >
            <div
              className="rdtTime"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      <div
                        className="rdtCounters"
                      >
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            2
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            23
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            16
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            PM
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter rdtMilli"
                        >
                          <input
                            onChange={[Function]}
                            type="text"
                            value="784"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <span
          aria-hidden="true"
          className="glyphicon glyphicon-time form-control-feedback"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker label should render a horizontal label with a specified width 1`] = `
<div
  className="custom-class"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="form-group clearfix "
  >
    <label
      className="control-label col-sm-6"
    >
      Default caption
    </label>
    <div
      className="col-sm-6"
    >
      <div
        className="widget-timepicker"
      >
        <div
          className="rdt"
        >
          <input
            className="form-control"
            disabled={false}
            name="custom-name"
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Placeholder"
            tabIndex={0}
            type="text"
            value="02:23:16.784 PM"
          />
          <div
            className="rdtPicker"
          >
            <div
              className="rdtTime"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      <div
                        className="rdtCounters"
                      >
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            2
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            23
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            16
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            PM
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter rdtMilli"
                        >
                          <input
                            onChange={[Function]}
                            type="text"
                            value="784"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <span
          aria-hidden="true"
          className="glyphicon glyphicon-time form-control-feedback"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker label should render a vertical label 1`] = `
<div
  className="custom-class"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="form-group clearfix "
  >
    <label
      className="control-label "
    >
      Default caption
    </label>
    <div
      className=""
    >
      <div
        className="widget-timepicker"
      >
        <div
          className="rdt"
        >
          <input
            className="form-control"
            disabled={false}
            name="custom-name"
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Placeholder"
            tabIndex={0}
            type="text"
            value="02:23:16.784 PM"
          />
          <div
            className="rdtPicker"
          >
            <div
              className="rdtTime"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      <div
                        className="rdtCounters"
                      >
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            2
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            23
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            16
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            PM
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter rdtMilli"
                        >
                          <input
                            onChange={[Function]}
                            type="text"
                            value="784"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <span
          aria-hidden="true"
          className="glyphicon glyphicon-time form-control-feedback"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker label should render with a different caption 1`] = `
<div
  className="custom-class"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="form-group clearfix "
  >
    <label
      className="control-label "
    >
      Different
    </label>
    <div
      className=""
    >
      <div
        className="widget-timepicker"
      >
        <div
          className="rdt"
        >
          <input
            className="form-control"
            disabled={false}
            name="custom-name"
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Placeholder"
            tabIndex={0}
            type="text"
            value="02:23:16.784 PM"
          />
          <div
            className="rdtPicker"
          >
            <div
              className="rdtTime"
            >
              <table>
                <tbody>
                  <tr>
                    <td>
                      <div
                        className="rdtCounters"
                      >
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            2
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            23
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            16
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounter"
                        >
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▲
                          </span>
                          <div
                            className="rdtCount"
                          >
                            PM
                          </div>
                          <span
                            className="rdtBtn"
                            onContextMenu={[Function]}
                            onMouseDown={[Function]}
                          >
                            ▼
                          </span>
                        </div>
                        <div
                          className="rdtCounterSeparator"
                        >
                          :
                        </div>
                        <div
                          className="rdtCounter rdtMilli"
                        >
                          <input
                            onChange={[Function]}
                            type="text"
                            value="784"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <span
          aria-hidden="true"
          className="glyphicon glyphicon-time form-control-feedback"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`TimePicker should render 24-hour notation 1`] = `
<div
  className="custom-class"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="widget-timepicker"
  >
    <div
      className="rdt"
    >
      <input
        className="form-control"
        disabled={false}
        name="custom-name"
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Placeholder"
        tabIndex={0}
        type="text"
        value="14:23:16.784"
      />
      <div
        className="rdtPicker"
      >
        <div
          className="rdtTime"
        >
          <table>
            <tbody>
              <tr>
                <td>
                  <div
                    className="rdtCounters"
                  >
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        14
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        23
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        16
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter rdtMilli"
                    >
                      <input
                        onChange={[Function]}
                        type="text"
                        value="784"
                      />
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-time form-control-feedback"
    />
  </div>
</div>
`;

exports[`TimePicker should render the structure 1`] = `
<div
  className="custom-class"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="widget-timepicker"
  >
    <div
      className="rdt"
    >
      <input
        className="form-control"
        disabled={false}
        name="custom-name"
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Placeholder"
        tabIndex={0}
        type="text"
        value="02:23:16.784 PM"
      />
      <div
        className="rdtPicker"
      >
        <div
          className="rdtTime"
        >
          <table>
            <tbody>
              <tr>
                <td>
                  <div
                    className="rdtCounters"
                  >
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        2
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        23
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        16
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        PM
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter rdtMilli"
                    >
                      <input
                        onChange={[Function]}
                        type="text"
                        value="784"
                      />
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-time form-control-feedback"
    />
  </div>
</div>
`;

exports[`TimePicker should render without a data attribute 1`] = `
<div
  className="custom-class"
  style={
    Object {
      "backgroundColor": "transparent",
    }
  }
>
  <div
    className="widget-timepicker"
  >
    <div
      className="rdt"
    >
      <input
        className="form-control"
        disabled={true}
        name="custom-name"
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="No attribute selected"
        tabIndex={0}
        type="text"
        value=""
      />
      <div
        className="rdtPicker"
      >
        <div
          className="rdtTime"
        >
          <table>
            <tbody>
              <tr>
                <td>
                  <div
                    className="rdtCounters"
                  >
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        12
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        00
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        00
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounter"
                    >
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▲
                      </span>
                      <div
                        className="rdtCount"
                      >
                        AM
                      </div>
                      <span
                        className="rdtBtn"
                        onContextMenu={[Function]}
                        onMouseDown={[Function]}
                      >
                        ▼
                      </span>
                    </div>
                    <div
                      className="rdtCounterSeparator"
                    >
                      :
                    </div>
                    <div
                      className="rdtCounter rdtMilli"
                    >
                      <input
                        onChange={[Function]}
                        type="text"
                        value="000"
                      />
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-time form-control-feedback"
    />
  </div>
</div>
`;
